src/main/groovy/io/seqera/wave/service/builder/ContainerBuildServiceImpl.groovy

```diff
@@ -259,9 +259,14 @@ class ContainerBuildServiceImpl implements ContainerBuildService {
         // since it was unable to initialise the build result status
         // this means the build status already exists, retrieve it
         final ret2 = buildStore.getBuild(request.targetImage)
-        if( ret2 ) {
+        if( ret2.getExitStatus() == null  || ret2.getExitStatus() == 0 ) {
             log.info "== Hit build cache for request: $request"
             return new BuildTrack(ret2.id, request.targetImage, true)
+        }else{
+            log.info "== Previous build for this image was failed, building again: $request"
+            buildStore.storeBuild(request.targetImage,ret1)
+            launchAsync(request)
+            return new BuildTrack(ret1.id, request.targetImage, false)
```

Comment:
The problem with this approach is that this block is not concurrency safe. This is why above if used `storeIfAbsent` that use an atomic operation to check and update the entry.

Reply:
I also think is not needed, because when a build fails the entry is stored with a shorter ttl to allow a new retry after a few second. See here 

https://github.com/seqeralabs/wave/blob/d81baa748762a05ceb750bfe5fc2b70be9458688/src/main/groovy/io/seqera/wave/service/builder/ContainerBuildServiceImpl.groovy#L199-L206

Reply:
Ok make sense